fecgov/fec-cms
# [DO NOT MERGE] 3673 Admin Fines Calc

## Summary

- Resolves #3673 
- Adds the admin fines calculator option to the base template of the site
- First PR for the admin fines calculator (calc af)

(Not _awful_ if it's merged, apart from the calculator _could_ be activated before the fine logic is in)

## Impacted areas of the application
- Added code to `base.html` to detect the calcaf class name and include its modal code if needed
- `calc-admin-fines.modal.js` runs this modal, based on other modals on the site
- `calc-admin-fines.js` runs the calculator itself
- `_calc-admin-fines.scss` are the styles specifically for the calculator
- Added ☝️ to `base.scss` so it's part of the base template (We may want to separate those before launch. Still undecided.)
- Added `calc-admin-fines.js` to the Webpack build

## Screenshots
![image](https://user-images.githubusercontent.com/26720877/80217043-0a13fb80-860d-11ea-8893-6e42e33e6656.png)


## Related PRs
None yet

## How to test
- Pull the branch
- `npm i`
- `npm run build`
- **Add the Vue CDN to the approved CORS domains**
  - Edit `fec/fec/middleware.py`, line 14 (the `script-src` entry)
  - Update `digitalgov.gov",` to `digitalgov.gov https://cdn.jsdelivr.net",`
  - Save your edit (after testing, feel free to discard this change)
  - No reason to commit this yet. That CDN and approach still TBD. Thoughts?
- `./manage.py runserver`
- Add a link to a page
  - In your 127.0.0.1 / localhost, go into [Wagtail](http://127.0.0.1:8000/admin) and edit a page—any page that uses the base.html template. I know that's hard to quickly find/decipher. I used `http://127.0.0.1:8000/admin/pages/7600/edit/`
  - Add some html somewhere: `<a href="" class="js-admin-fines-calc-modal">Launch the Administrative Fine Calculator now</a>` I added a new html block to the page and put my code in there. You can handle it however you like as long as the class is the same (may need to be an anchor tag but maybe not)
  - Save Draft, [Preview](http://127.0.0.1:8000/admin/pages/7600/edit/preview/)
  - Your edit / addition should be on the page
- Click your link and the calculator should open in a modal
- Try to break it!
- Is it clear?
- Is it easy? Fun?

## Known issues
- The fine logic/calculation isn't in yet. Right now, the final fine should always be 12345678.9
- A couple little style concerns (left border of help section doesn't reach the bottom, etc)
- I'm embarrassed to admit that I did not approach this as mobile-first; I still need to look at the smaller layouts
- For testing, frames don't auto advance (think they should?)
- After early testing, the Next button will become inactive until each frame has an answer
- Close and Start over buttons don't do anything
- Haven't even thought about this + Internet Explorer until this second

## Other considerations to test, but probably later
- How does the link behave if there are two on the page?
- Does this addition compete with other modals on the page?
- How does the functionality handle when the class is on something that's not a link?
- @johnnyporkchops suggested being able to add the calculator as a Wagtail module. _mi piace_. Interesting…


____
